From 2a5172da4aa933f70d312dec2aecdff2bdb8402b Mon Sep 17 00:00:00 2001
From: Spencer Sharkey <spencer@sf-n.com>
Date: Thu, 23 May 2019 00:45:00 -0700
Subject: [PATCH] Tick EntityInsentient less frequently


diff --git a/src/main/java/net/minecraft/server/EntityInsentient.java b/src/main/java/net/minecraft/server/EntityInsentient.java
index d1ec201d..b3bf5bbf 100644
--- a/src/main/java/net/minecraft/server/EntityInsentient.java
+++ b/src/main/java/net/minecraft/server/EntityInsentient.java
@@ -258,11 +258,17 @@ public abstract class EntityInsentient extends EntityLiving {
     }
 
     public void tick() {
+        // start uplol - Tick InsentientEntities every 5th tick
+        if (ticksLived > 2 && ticksLived % 5 == 0) {
+            return;
+        }
+        // end uplol
+
         super.tick();
         if (isTypeNotPersistent() && hasBeenCounted == this.isPersistent()) ((com.destroystokyo.paper.PaperWorldEntityList) this.world.entityList).updateEntityCount(this, hasBeenCounted ? -1 : 1); // Paper - adjust count if persistence state changes
         if (!this.world.isClientSide) {
             this.dl();
-            if (this.ticksLived % 5 == 0) {
+            if (this.ticksLived % 10 == 0) {
                 boolean flag = !(this.bO() instanceof EntityInsentient);
                 boolean flag1 = !(this.getVehicle() instanceof EntityBoat);
 
diff --git a/src/main/java/net/minecraft/server/EntityLiving.java b/src/main/java/net/minecraft/server/EntityLiving.java
index 4d5459d2..e3672965 100644
--- a/src/main/java/net/minecraft/server/EntityLiving.java
+++ b/src/main/java/net/minecraft/server/EntityLiving.java
@@ -2103,6 +2103,7 @@ public abstract class EntityLiving extends Entity {
         super.tick();
         this.cV();
         this.o();
+
         if (!this.world.isClientSide) {
             int i = this.getArrowCount();
 
-- 
2.17.1

